L1Trigger/Phase2L1GT/plugins/L1GTSingleCollectionCut.h

```diff
@@ -0,0 +1,170 @@
+      return res;
+    }
+
+    unsigned int atIndex(int objeta) const {
+      // Function that checks at which index the eta of the object is
+      // If object abs(eta) < regionsAbsEtaLowerBounds_[0] the function returns the last index,
+      // Might be undesired behaviour
+      for (unsigned int i = 0; i < regionsAbsEtaLowerBounds_.size() - 1; i++) {
+        if (std::abs(objeta) >= scales_.to_hw_eta(regionsAbsEtaLowerBounds_[i]) &&
+            std::abs(objeta) < scales_.to_hw_eta(regionsAbsEtaLowerBounds_[i + 1])) {
+          return i;
+        }
+      }
+      return regionsAbsEtaLowerBounds_.size() - 1;
+    }
+
+    bool checkObject(const P2GTCandidate& obj) const {
+      bool result = true;
+
+      result &= minPt_ ? (obj.hwPT() > minPt_) : true;
+      result &= maxPt_ ? (obj.hwPT() < maxPt_) : true;
+
+      result &= minEta_ ? (obj.hwEta() > minEta_) : true;
+      result &= maxEta_ ? (obj.hwEta() < maxEta_) : true;
+
+      result &= minPhi_ ? (obj.hwPhi() > minPhi_) : true;
+      result &= maxPhi_ ? (obj.hwPhi() < maxPhi_) : true;
+
+      result &= minZ0_ ? (obj.hwZ0() > minZ0_) : true;
+      result &= maxZ0_ ? (obj.hwZ0() < maxZ0_) : true;
+
+      result &= minEtaAbs_ ? (abs(obj.hwEta()) > minEtaAbs_) : true;
+      result &= maxEtaAbs_ ? (abs(obj.hwEta()) < maxEtaAbs_) : true;
+
+      result &= minScalarSumPt_ ? (obj.hwSca_sum() > minScalarSumPt_) : true;
+      result &= maxScalarSumPt_ ? (obj.hwSca_sum() < minScalarSumPt_) : true;
+
+      result &= qual_.empty() ? true : std::find(qual_.begin(), qual_.end(), obj.hwQual().to_uint()) != qual_.end();
+      result &=
+          maxIso_ ? obj.hwIso().to_int() << scales_.isolation_shift() < maxIso_.value() * obj.hwPT().to_int() : true;
+
+      result &= minHwIso_ ? (obj.hwIso() > minHwIso_) : true;
+      result &= regionsAbsEtaLowerBounds_.empty() ? true : checkEtadependentcuts(obj);
+      return result;
+    }
+
+    static void fillDescriptions(edm::ParameterSetDescription& desc) {
+      desc.add<edm::InputTag>("tag");
+      desc.addOptional<double>("minPt");
+      desc.addOptional<double>("maxPt");
+      desc.addOptional<double>("minEta");
+      desc.addOptional<double>("maxEta");
+      desc.addOptional<double>("minPhi");
+      desc.addOptional<double>("maxPhi");
+      desc.addOptional<double>("minZ0");
+      desc.addOptional<double>("maxZ0");
+      desc.addOptional<double>("minScalarSumPt");
+      desc.addOptional<double>("maxScalarSumPt");
+      desc.addOptional<std::vector<unsigned int>>("qual");
+      desc.addOptional<double>("minEtaAbs");
+      desc.addOptional<double>("maxEtaAbs");
+      desc.addOptional<double>("maxIso");
+      desc.addOptional<int>("minHwIso");
+      desc.addOptional<std::vector<double>>("regionsAbsEtaLowerBounds");
+      desc.addOptional<std::vector<double>>("regionsMinPt");
+      desc.addOptional<std::vector<double>>("regionsMaxIso");
+      desc.addOptional<std::vector<unsigned int>>("regionsQual");
```

Comment:
These could be required parameters with a default value to avoid the `exists()` calls in the constructor.